src/Analyzers/CSharp/Analyzers/UsePatternCombinators/AnalyzedPattern.cs

```diff
@@ -23,9 +25,44 @@ private AnalyzedPattern(IOperation target)
         /// </summary>
         internal sealed class Type : AnalyzedPattern
         {
+            public static Type? TryCreate(TypeSyntax typeSyntax, IIsTypeOperation operation)
+            {
+                Contract.ThrowIfNull(operation.SemanticModel);
+
+                // We are coming from a type pattern, which only binds to types, but converting to
```

Comment:
I think "*only*" is not accurate. It can bind to expressions if no type is found.